src/ecs/resources/input/input_handler.rs

```diff
@@ -81,6 +28,8 @@ impl<'a> Iterator for Buttons<'a> {
 /// ```
 #[derive(Default)]
 pub struct InputHandler {
```

Comment:
Doesn't this struct totally blow up the stack? Although you did this in another PR, I don't think we need `SmallVec` here since it's not like a collection of collections.